3-calendar-view-for-month

BetOnJeanette opened the pull request:
# What was the problem?
There was no monthly layout for the calendar

# How does this address the problem?
I added a header that shows the current month and year being shown, with links that move to the previous and next months.

I created a shared component for weekly layout, as well as the shared style to have a body evenly divided between 7 sections.

I also implemented the base layout, where each month has a filler space between sunday and the first day of the month (unless it starts on a sunday). Then, we fill the rest of the month with day components.

This closes #3.

MsZoezo commented on src/CalendarViews/SharedStyling/WeekFormating.css:
```diff
@@ -0,0 +1,5 @@
+.weekFormat {
+    display: grid;
+    grid-template-columns: calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7) calc(100% / 7);
```
while this works, there are built in ways to repeatedly assign fractions of the width to a column.

```css
grid-template-columns: repeat(7, 1fr);
```
this is cleaner to read and if you add something like a gap later on it'll handle it no problem too.

MsZoezo reviewed (changes requested):
Currently the .tsx files are also quite messy in the use (or lack of) ; at the end of lines.
(Something I seem to have missed in my previous review >~<)

Either way I do think it's important to keep being consistent with this, maybe we should look into a linter after all?

BetOnJeanette commented:
Updated the branch: 
- Used the CSS repeat (sorry, didn't know that was a thing)
- Pushed semicolons off (see #21)
- Destructured props
- Added JSDoc comments for everything

MsZoezo commented on src/CalendarViews/MonthlyCalendar/MainBody/StartOfMonthBuffer/StartOfMonthBuffer.tsx:
```diff
@@ -0,0 +1,21 @@
+ * @returns If the first day is a sunday, an empty element. 
+ *          Otherwise, a buffer element between sunday and the first date.
+ */
+function StartOfMonthBuffer(props: Props){
+    let { firstOfMonth } = props
```
```suggestion
function StartOfMonthBuffer({ firstOfMonth }: Props){
```

I would recommend destructuring like this, keeps it all in one line all neatly.

MsZoezo commented on src/CalendarViews/MonthlyCalendar/MainBody/SingleDate/SingleDate.tsx:
```diff
@@ -0,0 +1,16 @@
+ * @param props The necessary information to be passed in
+ * @returns A JSX element representing the day.
+ */
+function SingleDate(props: Props){
+    let { date } = props
```
```suggestion
function SingleDate({ date }: Props){
```

MsZoezo commented on src/CalendarViews/MonthlyCalendar/MainBody/MainBody.tsx:
```diff
@@ -0,0 +1,62 @@
+ * @param props The props used to recreate the element
+ * @returns A JSX copy of the body of the monthly view
+ */
+function CalendarBody(props: Props) {
+    let {date} = props
```
```suggestion
function CalendarBody({ date }: Props) {
```

MsZoezo commented on src/CalendarViews/MonthlyCalendar/Header/CalendarHeader.tsx:
```diff
@@ -0,0 +1,22 @@
+ * @param props The information necessary to create the header
+ * @returns a JSX element for the header 
+ */
+function CalendarHeader(props: Props) {
+    let {date, changeMonthYear} = props
```
```suggestion
function CalendarHeader({ date, changeMonthYear }: Props) {
```

MsZoezo reviewed (approved):
This code transcends time (which may be an issue when you deal with a calendar)🌠
